eth/vm/forks/london/transactions.py

```diff
@@ -0,0 +1,352 @@
+        return Receipt(
+            state_root=status,
+            gas_used=gas_used,
+            logs=logs,
+        )
+
+
+class BaseGasFeePayloadDecoder(TransactionDecoderAPI):
+    @classmethod
+    def decode(cls, payload: bytes) -> SignedTransactionAPI:
+        return rlp.decode(payload, sedes=BaseGasFeeTransaction)
+
+
+class LondonTypedTransaction(TypedTransaction):
+    decoders: Dict[int, Type[TransactionDecoderAPI]] = {
+        ACCESS_LIST_TRANSACTION_TYPE: AccessListPayloadDecoder,
+        BASE_GAS_FEE_TRANSACTION_TYPE: BaseGasFeePayloadDecoder,
+    }
+
+    def __init__(self, type_id: int, proxy_target: SignedTransactionAPI) -> None:
+        super().__init__(type_id, proxy_target)
+        self.max_priority_fee_per_gas = self._inner.max_priority_fee_per_gas
+        self.max_fee_per_gas = self._inner.max_fee_per_gas
+
+    # @property
+    # def max_priority_fee_per_gas(self) -> int:
+    #     return self._inner.max_priority_fee_per_gas
+
+    # @property
+    # def max_fee_per_gas(self) -> int:
+    #     return self._inner.max_fee_per_gas
+
+
+class LondonTransactionBuilder(BerlinTransactionBuilder):
+    legacy_signed = LondonLegacyTransaction
+    legacy_unsigned = LondonUnsignedLegacyTransaction
+
+    @classmethod
+    def new_unsigned_base_gas_price_transaction(
+            cls,
+            chain_id: int,
+            nonce: int,
+            max_priority_fee_per_gas: int,
+            max_fee_per_gas: int,
+            gas: int,
+            to: Address,
+            value: int,
+            data: bytes,
+            access_list: Sequence[Tuple[Address, Sequence[int]]],) -> LondonTypedTransaction:
+        transaction = UnsignedBaseGasFeeTransaction(
+            chain_id,
+            nonce,
+            max_priority_fee_per_gas,
+            max_fee_per_gas,
+            gas,
+            to,
+            value,
+            data,
+            access_list,
+        )
+        return transaction
+
+    @classmethod
+    def new_base_gas_price_transaction(
+            cls,
+            chain_id: int,
+            nonce: int,
+            max_priority_fee_per_gas: int,
+            max_fee_per_gas: int,
+            gas: int,
+            to: Address,
+            value: int,
+            data: bytes,
+            access_list: Sequence[Tuple[Address, Sequence[int]]],
+            y_parity: int,
+            r: int,
+            s: int) -> LondonTypedTransaction:
+        transaction = BaseGasFeeTransaction(
+            chain_id,
+            nonce,
+            max_priority_fee_per_gas,
+            max_fee_per_gas,
+            gas,
+            to,
+            value,
+            data,
+            access_list,
+            y_parity,
+            r,
+            s,
+        )
+        return LondonTypedTransaction(BASE_GAS_FEE_TRANSACTION_TYPE, transaction)
+
+def normalize_transaction(
+    transaction: SignedTransactionAPI
+) -> SignedTransactionAPI:
```

Comment:
This function modifies the transaction object rather than creating an instance of a possible new `LondonTypedTransaction` class so that both new (London-specific) and existing validation code can work on all transactions. However this means that `max_priority_fee_per_gas` and `max_fee_per_gas` cannot be read-only properties, and I'm wondering how should we handle users trying to access the new properties in pre-London transactions?